tests/unit/use-machine-test.js

```diff
@@ -831,4 +831,213 @@ module('Unit | use-machine', function (hooks) {
+        );
+
+        this.set('state', 'active');
+
+        assert.equal(
+          this.test.statechart.state.value,
+          'active',
+          'state was changed from the outside interpreter was restarted with new state'
+        );
+      });
+
+      test('it is possible to restart the interpreter in a specific state within `update`', async function (assert) {
+        const testContext = this;
+
+        const machine = Machine({
+          id: 'wizard',
+          initial: 'step-a',
+          states: {
+            'step-a': {
+              on: {
+                NEXT: 'step-b',
+              },
+            },
+            'step-b': {
+              on: {
+                NEXT: 'step-c',
+                PREV: 'step-b',
+              },
+            },
+            'step-c': {
+              on: {
+                PREV: 'step-b',
+              },
+            },
+          },
+        });
+
+        class Test extends Component {
+          @use statechart = useMachine(machine)
+            .withContext({
+              currentWizardStep: this.args.step,
+            })
+            .update(({ restart, context: { currentWizardStep } }) => {
+              restart(currentWizardStep);
+            });
+
+          constructor(owner, args) {
+            super(owner, args);
+
+            testContext.test = this;
+          }
+        }
+
+        this.owner.register('component:test', Test);
+
+        await render(hbs`
+          <Test @step={{this.step}}/>
+        `);
+
+        assert.equal(
+          this.test.statechart.state.value,
+          'step-a',
+          'no state passed initially - current state is the correct initial state'
+        );
+
+        this.set('step', 'step-c');
+
+        assert.equal(
+          this.test.statechart.state.value,
+          'step-c',
+          'state was changed from the outside interpreter was restarted with new state'
+        );
+      });
+    });
+  });
+
+  module('#onTransition', function () {
+    test('it is possible to react to state changes', async function (assert) {
+      const testContext = this;
+
+      const machine = Machine({
+        initial: 'inactive',
+        states: {
+          inactive: {
+            on: {
+              START: 'active',
+            },
+          },
+          active: {
+            on: {
+              STOP: 'inactive',
+            },
+          },
+        },
+      });
+
+      class Test extends Component {
+        @use statechart = useMachine(machine).onTransition((state) => {
+          assert.step(state.value);
+        });
+
+        constructor(owner, args) {
+          super(owner, args);
+
+          testContext.test = this;
+        }
+      }
+
+      this.owner.register('component:test', Test);
+
+      await render(hbs`
+        <Test @state={{this.state}}/>
+      `);
+
+      this.test.statechart.send('START');
+
+      assert.verifySteps(['inactive', 'active'], 'all state transitions fired `onTransition`');
```

Comment:
Maybe we should split this up into 2 assertions, so we know `onTransition` is called for the initial transition:

```js
this.test.statechart;

assert.verifySteps(['inactive']);

this.test.statechart.send('START');

assert.verifySteps(['active']);
```

?

Reply:
done